[CI] MistralServiceTests testUnifiedCompletionStreamingNotFoundError failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #12322 / windows-2022_checkpart5_platform-support-windows](https://gradle-enterprise.elastic.co/s/olwuvlvl6jkd6)

**Reproduction Line:**
```
gradlew ":x-pack:plugin:inference:test" --tests "org.elasticsearch.xpack.inference.services.mistral.MistralServiceTests.testUnifiedCompletionStreamingNotFoundError" -Dtests.seed=21E7F71A397B7E0F -Dtests.locale=ff-Latn -Dtests.timezone=Europe/Zagreb -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.inference.services.mistral.MistralServiceTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testUnifiedCompletionStreamingNotFoundError'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: is "{\"error\":{\"code\":\"not_found\",\"message\":\"Resource not found at [http://127.0.0.1:60473] for request from inference entity id [id] status [404]. Error message: [{\\n    \\\"detail\\\": \\\"Not Found\\\"\\n}\\n]\",\"type\":\"mistral_error\"}}"
     but: was "{\"error\":{\"code\":\"not_found\",\"message\":\"Resource not found at [http://127.0.0.1:60473] for request from inference entity id [id] status [404]\",\"type\":\"mistral_error\"}}"
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testUnifiedCompletionStreamingNotFoundError
- [main] 1 consecutive failures in step windows-2022_checkpart5_platform-support-windows
- [main] 1 consecutive failures in pipeline elasticsearch-periodic-platform-support

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -764,9 +764,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.ndjson.NdJsonCompressedFormatSpecIT
   method: test {csv-spec:external-basic.aggregateCount [ndjson.gz/S3]}
   issue: https://github.com/elastic/elasticsearch/issues/150710
-- class: org.elasticsearch.xpack.inference.services.mistral.MistralServiceTests
-  method: testUnifiedCompletionStreamingNotFoundError
-  issue: https://github.com/elastic/elasticsearch/issues/150711
 - class: org.elasticsearch.xpack.core.ml.datafeed.DatafeedConfigTests
   method: testEqualsAndHashcode
   issue: https://github.com/elastic/elasticsearch/issues/150712

--- a/x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/mistral/MistralServiceTests.java
+++ b/x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/mistral/MistralServiceTests.java
@@ -288,7 +288,7 @@ public class MistralServiceTests extends InferenceServiceTestCase {
         }
     }
 
-    public void testUnifiedCompletionStreamingNotFoundError() throws Exception {
+    public void testUnifiedCompletionNonStreamingNotFoundError() throws Exception {
         String responseJson = """
             {
                 "detail": "Not Found"

